utils/analysis/helpers.go

```diff
@@ -328,6 +329,41 @@ func flattenMetrics(templates []*v1alpha1.AnalysisTemplate, clusterTemplates []*
 	return metrics, nil
 }
 
+func NewAnalysisRunFromUnstructured(obj *unstructured.Unstructured, templateArgs []v1alpha1.Argument, name, generateName, namespace string) (*unstructured.Unstructured, error) {
+	var newArgs []v1alpha1.Argument
+	var un unstructured.Unstructured
+
+	objArgs, notFound, err := unstructured.NestedSlice(un.Object, "metadata", "args")
+	if err != nil {
+		return nil, err
+	}
+	if notFound {
+		newArgs = templateArgs
+	} else {
+		// AnalysisTemplate has arguments set
+		var args []v1alpha1.Argument
+		argBytes, err := json.Marshal(objArgs)
+		yaml.Unmarshal(argBytes, &args)
+		newArgs, err = MergeArgs(templateArgs, args)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+
+	// Change kind to AnalysisRun
+	unstructured.SetNestedField(obj.Object, "AnalysisRun", "kind")
+
+	unstructured.SetNestedField(obj.Object, name, "metadata", "name")
+	unstructured.SetNestedField(obj.Object, generateName, "metadata", "generateName")
+	unstructured.SetNestedField(obj.Object, namespace, "metadata", "namespace")
+
+	// Set args
+	unstructured.SetNestedField(obj.Object, newArgs, "spec", "args")
```

Comment:
@jessesuen This line is causing an error: ```panic: cannot deep copy *[]v1alpha1.Argument```

Reply:
yea you have to convert []v1alpha1.Argument to map[string]interface